### PR TITLE
Fix delta_changes_field_changed check for Rails 5

### PR DESCRIPTION
**Pull request by labe**
In Rails 4.2, `#delta_changes_field_changed?` compares the currently written attribute value to a passed-in argument value (`old`); in Rails 5, the currently written value is compared to the current database value.

This difference in behavior is significant in that it prevents use of `#delta_changes` as a way to log requested changes to a tracked column within a single database transaction, in the event where the final written value matches what was already in the database.

An example use case has been added to the test file (and the models have been appropriately expanded) to demonstrate a context and flow wherein the current code fails and the proposed change succeeds.

### CODE REVIEW

**Comment by labe** [comment]
Darn it. This seems to increase R5 test failures... via the GH script:
```
Github PR# 46723 for R5U-1836_fix_ticket_integrity_tests
{:ID=>536252723}
2021-02-04 00:36:04 -0600
Fetching and extracting logs zip file...
...saving cache
Aggregating...
4: 0 errors, 0 failures, 0 files
5: 1 errors, 174 failures, 89 files
+: 1 errors, 174 failures, 89 files

Branch master
{:ID=>536045995}
2021-02-03 22:44:47 -0600
...loading from cache
Aggregating...
4: 0 errors, 0 failures, 0 files
5: 0 errors, 163 failures, 88 files
+: 0 errors, 163 failures, 88 files
```

**Comment by labe** [comment]
Current status 😅 :

```
Aggregating...
4: 0 errors, 0 failures, 0 files
5: 0 errors, 160 failures, 86 files
+: 0 errors, 160 failures, 86 files

## Changes from master:

-test/lib/zendesk/custom_field/custom_field_manager_test.rb:217
-test/models/ticket/integrity_test.rb:466
-test/models/ticket/integrity_test.rb:650
```

**Review by zenspider — APPROVED** [review]
This doesn't have rails' odd internal casting thing... but I doubt it matters much in this case.

**Review by grosser — COMMENTED** [review]
add a test-case that fails without this ?

File: Gemfile.rails5 [hunk 1]
```diff
@@ -0,0 +1 @@
+eval_gemfile('gemfiles/rails5.0.gemfile')
```
Comment (grosser) [inline Gemfile.rails5:1]: remove this ? ... can `BUNDLE_GEMFILE=gemfiles/rails5.0.gemfile`

File: Readme.md [hunk 2]
```diff
@@ -30,6 +30,16 @@ user.reset_delta_changes!
 user.delta_changes # => {}
 ```
 
+Testing
+======
+
+To run tests with Rails 4.2: `$ rake spec`
```
Comment (grosser) [inline Readme.md:36]: ideally the Gemfile should be the same as latest ... so `eval_gemfile Dir["gemfiles/*.gemfile"].sort.last`

File: Readme.md [hunk 3]
```diff
@@ -30,6 +30,16 @@ user.reset_delta_changes!
...
+To run tests with Rails 5.0:
+```
+$ BUNDLE_GEMFILE=Gemfile.rails5 bundle
```
Comment (grosser) [inline Readme.md:39]:
```suggestion
$ BUNDLE_GEMFILE=gemfiles/rails5.0.gemfile bundle
```

File: spec/delta_changes_spec.rb [hunk 4]
```diff
@@ -103,5 +103,24 @@
       user.score = '5'
       expect(user.delta_changes).to eq({})
     end
+
+    context 'when a tracked column is changed multiple times in a single db transaction' do
+      context 'and the final value is the same as the original' do
+        context 'and delta_changes is reset between each change' do
+          it 'should be filled after each change is applied' do
```
Comment (grosser) [inline spec/delta_changes_spec.rb:110]: can leave all the context in a comment if it is important
```suggestion
    it 'does not track extra changes when changing back and forth from original value'
```

**Review by grosser — APPROVED** [review]

**Review by saturnflyer — APPROVED** [review]

**Pull request merged**